casa/IO/test/tByteSink.cc

```diff
@@ -107,9 +107,10 @@ int main ()
 	uChar    testuChar;
 	String   testString;
 
-	RegularFileIO regularFileIO (Path("tByteSink_tmp.dat"));
-	CanonicalIO canonicalIO(&regularFileIO);
-	ByteSource  source(&canonicalIO);
+        std::shared_ptr<ByteIO> regularFileIO
+          (new RegularFileIO (Path("tByteSink_tmp.dat")));
+        std::shared_ptr<TypeIO> canonicalIO (new CanonicalIO(regularFileIO));
```

Comment:
```suggestion
        std::shared_ptr<RegularFileIO> regularFileIO
          (new RegularFileIO (Path("tByteSink_tmp.dat")));
        std::shared_ptr<CanonicalIO> canonicalIO (new CanonicalIO(regularFileIO));
```
Additionally, you could use `std::make_shared` here.